src/lsm/composite_key.zig

```diff
@@ -4,16 +4,16 @@ const math = std.math;
 
 const stdx = @import("../stdx.zig");
 
-pub fn CompositeKey(comptime Field: type) type {
+pub fn CompositeKeyType(comptime Field: type) type {
     assert(Field == u128 or Field == u64);
 
     return extern struct {
```

Comment:
Packed struct simplifies the alignment, and `packet struct(IntType)` communicates the intent nicely.

Also:
- Added asserts for `padding == pad`
- Added asserts for `timestamp & tombstone_bit == 0`
- Padding now can be `void` instead of `[0]u8`